packages/contractkit/src/wallets/signers/azure-key-vault-client.ts

```diff
@@ -0,0 +1,191 @@
+import { ec as EC } from 'elliptic'
+import * as ethUtil from 'ethereumjs-util'
+import { ecdsaRecover } from 'secp256k1'
+
+/**
+ * Provides an abstraction on Azure Key Vault for performing signing operations
+ */
+export class AzureKeyVaultClient {
+  private readonly vaultName: string
+  // Unique URI of the Azure Key Vault
+  private readonly vaultUri: string
+  private readonly credential: DefaultAzureCredential
+  private readonly keyClient: KeyClient
+  private readonly SIGNING_ALGORITHM: string = 'ECDSA256'
+
+  // 0x04 prefix indicates that the key is not compressed
+  // https://tools.ietf.org/html/rfc5480#section-2.2
+  private readonly publicKeyPrefix: number = 0x04
+
+  constructor(vaultName: string) {
+    this.vaultName = vaultName
+    this.vaultUri = `https://${this.vaultName}.vault.azure.net`
+    // DefaultAzureCredential supports service principal or managed identity
+    // If using a service principal, you must set the appropriate environment vars
+    this.credential = new DefaultAzureCredential()
+    this.keyClient = new KeyClient(this.vaultUri, this.credential)
+  }
+
+  public async getKeys(): Promise<string[]> {
+    const keyNames = new Array<string>()
+    for await (const keyProperties of this.keyClient.listPropertiesOfKeys()) {
+      keyNames.push(keyProperties.name)
+    }
+    return keyNames
+  }
+
+  public async getPublicKey(keyName: string): Promise<BigNumber> {
+    const signingKey = await this.getKey(keyName)
+
+    const pubKeyPrefix = Buffer.from(new Uint8Array([this.publicKeyPrefix]))
+    const rawPublicKey = Buffer.concat([
+      pubKeyPrefix,
+      Buffer.from(signingKey.key!.x!),
+      Buffer.from(signingKey.key!.y!),
+    ])
+    const publicKey = AzureKeyVaultClient.bufferToBigNumber(rawPublicKey)
+    return publicKey
+  }
+
+  public async getKeyId(keyName: string): Promise<string> {
+    if (!(await this.hasKey(keyName))) {
+      throw new Error(`Unable to locate key: ${keyName}`)
+    }
+    return (await this.getKey(keyName)).id!
+  }
+
+  public async signMessage(message: Buffer, keyName: string): Promise<Signature> {
+    if (!(await this.hasKey(keyName))) {
+      throw new Error(`Unable to locate key: ${keyName}`)
+    }
+    const keyId = await this.getKeyId(keyName)
+    const cryptographyClient = new CryptographyClient(keyId, this.credential)
+    // @ts-ignore-next-line (ECDSA256 is not included in the client enum but is valid)
+    const signResult = await cryptographyClient.sign(this.SIGNING_ALGORITHM, message)
+    // The output of this will be a 64 byte array.
+    // The first 32 are the value for R and the rest is S
+    if (
+      typeof signResult === 'undefined' ||
+      typeof signResult.result === 'undefined' ||
+      signResult.result.length !== 64
+    ) {
+      throw new Error(`Invalid signature returned from Azure: ${signResult}`)
+    }
+    const rawSignature = signResult.result
+
+    // Canonicalize signature
+    const secp256k1Curve = new EC('secp256k1')
```

Comment:
Looked briefly at `elliptic` to try to understand it a little better and they recommend creating this once and reusing if you can

Reply:
Thanks for noticing this!